lasagne/layers/recurrent.py

```diff
@@ -547,16 +547,24 @@ def __init__(self, incoming, num_units,
                  mask_input=None,
                  **kwargs):
         input_shape = incoming.output_shape
+        # Retrieve the supplied name, if it exists; otherwise use ''
+        if 'name' in kwargs:
+            basename = kwargs['name'] + '.'
+            del kwargs['name']
```

Comment:
You shouldn't delete the name before calling the super constructor, though. Either don't pass `**kwargs` to `DenseLayer` at all, or create a copy of `kwargs` without `'name'` to be passed on to `DenseLayer`.